llm/generate/gen_common.sh

```diff
@@ -39,8 +39,39 @@ init_vars() {
     *)
         ;;
     esac
-    if [ -z "${CMAKE_CUDA_ARCHITECTURES}" ] ; then 
-        CMAKE_CUDA_ARCHITECTURES="50;52;61;70;75;80"
+    if [ -f /etc/nv_tegra_release ] ; then
+        TEGRA_DEVICE=1
+        echo "Tegra device detected: ${JETSON_MODEL}, L4T Version ${JETSON_L4T}, Jetpack Version ${JETSON_JETPACK}"
+    fi
+    if [ -z "${CMAKE_CUDA_ARCHITECTURES}" ] ; then
+        if [ -n ${TEGRA_DEVICE} ]; then
+            echo "CMAKE_CUDA_ARCHITECTURES unset, values are needed for Tegra devices. Using default values defined at https://github.com/dusty-nv/jetson-containers/blob/master/jetson_containers/l4t_version.py"
+        fi
+        # Tegra devices will fail generate unless architectures are set:
```

Comment:
To clarify this comment, when you `go generate ./...` on a Jetson, it breaks if you try to target the wrong architectures?

For our official builds on ARM, we leverage Docker Desktop running on ARM Mac's, so my hope was we'd be able to generate an ARM binary that was inclusive of all the architectures and run on the different JetPack systems.

Reply:
Yeah this is what happens. It seems to be due to partial architecture support included in the Jetsons.

[default_arch_build_jetson.log](https://github.com/ollama/ollama/files/14315144/default_arch_build_jetson.log)

Reply:
That's unfortunate that don't include enough headers/def's to be able to target other families from a given device.  It sounds like there's still hope our official build could work, but uesrs building from source on device will need to narrow scope